aws/lambda_function.py

```diff
@@ -0,0 +1,174 @@
+# Connect to database, query for only rows with no predicted exit destination, 
+# 'wrangle' rows to be fit for modeling, then sending to an S3 bucket
+
+import sys
+import logging
+import psycopg2
+import pandas as pd
+import boto3
+import os
+
+# rds settings
+rds_host = os.environ.get('RDS_HOST')
+rds_username = os.environ.get('RDS_USERNAME')
+rds_user_pwd = os.environ.get('RDS_USER_PWD')
+
+# s3 settings
+s3 = boto3.resource('s3')
+s3_bucket = os.environ.get('S3_BUCKET')
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+try:
+    conn = psycopg2.connect(
+        host=rds_host,
+        user=rds_username,
+        password=rds_user_pwd)
+except:
+    logger.error("ERROR: Could not connect to Postgres instance.")
+    sys.exit()
+
+logger.info("SUCCESS: Connection to RDS Postgres instance succeeded")
+
+def wrangle(df):
+    print(f"Original shape: {df.shape}")
+    clean_df = df.copy()
+    # NOT CURRENT IN GUESTS DB
+    # columns = ["private_disability_income",'predicted_exit_destination']
+    clean_df.columns = map(str.lower, clean_df.columns)
+    clean_df.drop(columns="index", inplace=True)
+    # Drop sensitive columns
+    columns = ['enrollment_created_by',	'first_name',	'last_name',	
+                'ssn', 'dob']
+    clean_df.drop(columns=columns, inplace=True)
+    # drop reviewed columns
+    drop_col = ["days_enrolled_in_project","bednights_during_report_period", 
+                "entire_episode_bednights", "contact_services", 
+                "non-cash_count_at_exit", "info_release_status",
+                "other_public", "unemployement_income", "housing_checkins"]
+    clean_df.drop(columns=drop_col, inplace=True)
+    # Drop questionable columns
+    # REMINDER: important, missing values, these should be required
+    columns = ["zip", "income_at_entry", "income_at_exit", 
+            "length_of_time_homeless", "homeless_start_date",
+            "vet_status", "living_situation", "length_of_stay",
+            "homeless_start_date", "times_homeless_last_3years",
+            "total_months_homeless", "last_perm_address", "state",
+            "municipality", "housing_status", "domestic_violence",
+            "currently_fleeing", "when_dv_occured", "engagement_date",
+            "last_grade_completed", "school_status", "employed_status",
+            "reason_not_employed", "type_of_employment", "looking_for_work",
+            "soar_eligibility", "alcohol_abuse", "chronic_health_condition",
+            "developmental_disability", "substance_abuse",
+            "mental_health_problem", "physical_disability"]
+    clean_df.drop(columns=columns, inplace=True)
+    # Drop interesting most_recent columns, check in with @J, future feature engineering
```

Comment:
These 'most recent' columns are the result of some meetings with the city as a means to help track down previous guests with unknown exit destinations.  For example, a current guest of ours should have "FPS--ES--Open Doors" in the "most_recent_es" column, since their most recent Emergency Shelter enrollment was with us at Open Doors.  

This allows us to know if a previous guest with an unknown exit possibly left to another program by checking for the most current 'most recent' after the date of their exit with us.